sql.go

```diff
@@ -74,6 +74,11 @@ func (u *NullUUID) Scan(src interface{}) error {
 		return nil
 	}
 
+	if id, ok := src.(UUID); ok {
```

Comment:
On the code path at line 84, we delegate to `UnmarshalText` or `UnmarshalBinary`, which validate the UUID. Here, we do no such thing, and we take it for granted that if `src` is of type `UUID`, then it represents a valid UUID.

In the context of your ORM and its usage, is there ever a world in which this `src` is not a valid UUID?

Reply:
In any case, I think we should leave a comment here, which describes why we are doing this. It should mention that certain ORMs use `sql.Scanner` to convert Go values to other Go values.